.github/workflows/ci_benchmark.yml

```diff
@@ -15,7 +15,7 @@ permissions:
 
 jobs:
   benchmark:
-    if: github.repository == 'astropy/astropy' && contains(github.event.pull_request.labels.*.name, 'benchmark')
+    if: (github.repository == 'astropy/astropy' && ((github.event_name == 'pull_request' && github.event.action != 'labeled' && contains(github.event.pull_request.labels.*.name, 'benchmark')) || (github.event_name == 'pull_request' && github.event.action == 'labeled' && github.event.label.name == 'benchmark')))
```

Comment:
I guess this can be simplified by taking out `event_name == 'pull_request'`, which is shared between two (though it doesn't really matter). I.e.,
```
    if: (github.repository == 'astropy/astropy' && github.event_name == 'pull_request' && ((github.event.action != 'labeled' && contains(github.event.pull_request.labels.*.name, 'benchmark') || (github.event.action == 'labeled' && github.event.label.name == 'benchmark')))
```

Reply:
Now that it works, I'd prefer not to touch it further lol. And might also guard against future changes if we are explicit like this, because I cannot predict what else we will throw into this workflow. 🙈  Thanks for the review!